src/wp-includes/blocks.php

```diff
@@ -1250,14 +1250,18 @@ function unregister_block_style( $block_name, $block_style_name ) {
  * @since 5.8.0
  *
  * @param WP_Block_Type $block_type    Block type to check for support.
- * @param array         $feature       Path to a specific feature to check support for.
+ * @param string|array  $feature       Path to a specific feature to check support for.
  * @param mixed         $default_value Optional. Fallback value for feature support. Default false.
  * @return bool Whether the feature is supported.
  */
 function block_has_support( $block_type, $feature, $default_value = false ) {
 	$block_support = $default_value;
 	if ( $block_type && property_exists( $block_type, 'supports' ) ) {
-		$block_support = _wp_array_get( $block_type->supports, $feature, $default_value );
+		if ( is_array( $feature ) ) {
+			$block_support = _wp_array_get( $block_type->supports, $feature, $default_value );
+		} else {
+			$block_support = isset( $block_type->supports[ $feature ] ) ? $block_type->supports[ $feature ] : $default_value;
+		}
```

Comment:
I think we should also check the length of the array, e.g. something like

```php
if ( is_array( $feature ) && count( $feature ) === 1 ) {
	$feature = $feature[0];
}
```

If someone passes an array with just 1 element in it we can also short-circuit.

This way we don't necessarily have to change all existing instances in core, but can keep them as is.

And plugins and themes doing it wrong also benefit.

Reply:
I have included this check. Did a small rearrangement of condition to include this. w.r.t to core I have updated all the single element array to string.